backend/src/memos/memos.service.ts

```diff
@@ -1,10 +1,118 @@
-import { Injectable } from '@nestjs/common'
+import {
+  BadRequestException,
+  ForbiddenException,
+  GoneException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common'
+import { Template, TemplateVersion, User, Memo } from 'database/entities'
+import { randomBytes } from 'crypto'
+import { isTemplateIssuer } from 'templates/templates.util'
+import { Connection } from 'typeorm'
+import { TemplateStatus } from 'types'
+
+import { CreateMemoDto, CreateMemoResponseDto } from './dto'
 
 @Injectable()
 export class MemosService {
-  async createMemo(): Promise<void> {
-    return
+  constructor(private connection: Connection) {}
+
+  // Creates a new memo
+  async createMemo(
+    issuer: User,
+    _data: CreateMemoDto,
+  ): Promise<CreateMemoResponseDto> {
+    const { templateId, versionId, uin, uinType, params } = _data
+    /*
+    1. Check template status
+      a. If Public, OK
+      b. If Private, check if user is Issuer
+      c. If Archived, fail
```

Comment:
The definition of template statuses is here: https://github.com/opengovsg/memos/pull/43#discussion_r784501343, will need to update your conditions below to match.